src/librados/IoCtxImpl.cc

```diff
@@ -637,17 +635,25 @@ int librados::IoCtxImpl::writesame(const object_t& oid, bufferlist& bl,
 }
 
 int librados::IoCtxImpl::operate(const object_t& oid, ::ObjectOperation *o,
-				 ceph::real_time *pmtime, int flags, const jspan_context* otel_trace)
+				 ceph::real_time *pmtime, int flags,
+				 const jspan_context& otel_ctx)
 {
+  auto trace = client->tracer().add_span("IoCtx::operate", otel_ctx);
```

Comment:
why do we add a new span here?
also, adding the additional info on the status of the operation probbaly needs to go into its owne commit. as it is unrelated to the change in APIs

Reply:
i see now that this is to be more consistent with other functiosn that also does that